src/Microsoft.Identity.Web.MicrosoftGraph/TokenAcquisitionCredentialProvider.cs

```diff
@@ -30,6 +30,12 @@ public TokenAcquisitionCredentialProvider(ITokenAcquisition tokenAcquisition, IE
         /// <returns>A Task (as this is an async method).</returns>
         public async Task AuthenticateRequestAsync(HttpRequestMessage request)
         {
+            // In case of a retry there is already an authorization header; replace it with new one to avoid using expired token
+            if (request.Headers.Contains(Constants.Authorization))
```

Comment:
What about the following?

```CSharp
request.Headers[Constants.Authorization] = string.Format(
                    CultureInfo.InvariantCulture,
                    "{0} {1}",
                    Constants.Bearer,
                    await _tokenAcquisition.GetAccessTokenForUserAsync(_initialScopes).ConfigureAwait(false)));
```

Reply:
the HttpHeaders doesn't have a indexer 

![image](https://user-images.githubusercontent.com/3148438/95662455-5dd0bd80-0b37-11eb-8e5f-4a83665477ec.png)

Reply:
Ok. Thanks @davesmits